src/rest/impl/operations.service.ts

```diff
@@ -100,152 +101,93 @@ export default class OperationsServiceImpl extends OperationsService {
-    if (!sap.DeliveryPlaces) {
-      this.logger.error("Failed to read SAP item groups");
-      return;
-    }
+  private async readSapImportDeliveryPlaces() {
+    try {
+      const sapDeliveryPlacesService = SapServiceFactory.getDeliveryPlacesService();
+      const deliveryPlaces = await sapDeliveryPlacesService.listDeliveryPlaces();
+      const operationReport = await models.createOperationReport(OPERATION_SAP_DELIVERY_PLACE_SYNC);
 
-    const deliveryPlaces = sap.DeliveryPlaces.DeliveryPlaces;
-    if (!deliveryPlaces) {
-      this.logger.error("Failed to read SAP delivery places list");
+      deliveryPlaces.forEach((deliveryPlace) => {
+        tasks.enqueueSapDeliveryPlaceUpdate(operationReport.id, deliveryPlace);
+      });
+  
+      return operationReport;
+    } catch (e) {
+      this.logger.error(`Failed to read SAP item groups. error: ${e}`);
       return;
     }
-
-    const operationReport = await models.createOperationReport(OPERATION_SAP_DELIVERY_PLACE_SYNC);
-    deliveryPlaces.forEach((deliveryPlace) => {
-      tasks.enqueueSapDeliveryPlaceUpdate(operationReport.id, deliveryPlace);
-    });
-
-    return operationReport;
   }
 
   /**
-   * Reads item groups from sap and fills related task queue with data from file
-   * 
-   * @param {Object} sap SAP data object 
+   * Reads item groups from sap and fills related task queue with data
    */
-  private async readSapImportItemGroups(sap: SAPExportRoot) {
-    if (!sap.ItemGroups) {
-      this.logger.error("Failed to read SAP item groups");
-      return;
-    }
+  private async readSapImportItemGroups() {
+    try {
+      const sapItemGroupsService = SapServiceFactory.getItemGroupsService();
+      const itemGroups = await sapItemGroupsService.listItemGroups();
+      const operationReport = await models.createOperationReport("SAP_ITEM_GROUP_SYNC");
 
-    const itemGroups = sap.ItemGroups.ItemGroup;
-    if (!itemGroups) {
-      this.logger.error("Failed to read SAP item group list");
+      itemGroups.forEach((itemGroup) => {
+        tasks.enqueueSapItemGroupUpdate(operationReport.id, itemGroup);
+      });
+
+      return operationReport;
+    } catch (e) {
+      this.logger.error(`Failed to read SAP item groups. error: ${e}`);
       return;
     }
-
-    const operationReport = await models.createOperationReport("SAP_ITEM_GROUP_SYNC");
-
-    itemGroups.forEach((itemGroup) => {
-      tasks.enqueueSapItemGroupUpdate(operationReport.id, itemGroup);
-    });
-
-    return operationReport;
   }
 
   /**
-   * Reads contracts from SAP and fills related task queue with data from file
-   * 
-   * @param {Object[]} sapDatas Array of SAP data objects
+   * Reads contracts from SAP and fills related task queue with data
    */
-  private async readSapImportContracts(sapDatas: SAPData[]) {
+  private async readSapImportContracts() {
+    const sapContractsService = SapServiceFactory.getContractsService();
```

Comment:
Add proper error handling

Reply:
Added missing try-catch, but same here as in other SAP data import operations